scripts/parse_vcgs_manifest.py

```diff
@@ -0,0 +1,381 @@
+                # it already exists
+                samples_to_update[sample_name] = {
+                    'id': external_id_map[sample_name],
+                    'external_id': sample_name,
+                    'type': 'blood',
+                    'meta': collapsed_sample_meta
+                }
+            else:
+                samples_to_add.append(
+                    NewSample(
+                        external_id=sample_name,
+                        type=SampleType('blood'),
+                        meta=collapsed_sample_meta,
+                    )
+                )
+                sequencing_to_add[sample_name].append(
+                    NewSequence(
+                        sample_id='<None>', # keep the type initialise happy
+                        meta=collapsed_sequencing_meta,
+                        type=SequenceType(collapsed_type),
+                        status=SequenceStatus('uploaded'),
+                    )
+                )
+
+        if not self.sample_metadata_project:
+            logger.info('No sample-metadata project set, so skipping add into SM-DB')
+            return samples_to_add, sequencing_to_add
+
+
+        logger.info(f'Adding {len(samples_to_add)} samples to SM-DB database')
+        external_sample_id_to_internal_id = {}
+        for new_sample in samples_to_add:
+            sample_id = sapi.create_new_sample(
+                project=self.sample_metadata_project, new_sample=new_sample
+            )
+            external_sample_id_to_internal_id[new_sample.external_id] = sample_id
+
+        for sample_id, sequences_to_add in sequencing_to_add.items():
+            for seq in sequences_to_add:
+                seq.sample_id = external_sample_id_to_internal_id[sample_id]
+                seqapi.create_new_sequence(
+                    project=self.sample_metadata_project, new_sequence=seq
+                )
+
+        return external_sample_id_to_internal_id
+
+    def parse_sequencing_type(self, sample_id: str, types: List[str]):
+        # filter false-y values
+        types = list(set(t for t in types if t))
+        if len(types) <= 0:
+            return self.default_sequencing_type
+        if len(types) > 1:
+            raise ValueError(
+                f'Multiple library types for same sample {sample_id}, '
+                f'maybe there are multiples types of sequencing in the same '
+                f'manifest? If so, please raise an issue with mfranklin to '
+                f'change the groupby to include {Columns.LIBRARY_STRATEGY}.'
+            )
+
+        type_ = types[0].lower()
+        if type_ == 'wgs':
+            return 'wgs'
+        if type_ in ('single-cell', 'ss'):
+            return 'single-cell'
+
+        raise ValueError(f'Unrecognised sequencing type {type_}')
+
+    def parse_reads(self, reads: List[str]) -> Union[List[Dict], Dict]:
+        if all(r.endswith('.fastq') or r.endswith('.fastq.gz') for r in reads):
```

Comment:
Not relevant for this specific manifest, but to make the code reusable in future, can also check for the `.fq.gz` common extention